src/composables/useActiveSession.js

```diff
@@ -0,0 +1,132 @@
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { computed, onBeforeMount, onBeforeUnmount, ref, watch } from 'vue'
+
+import { getCapabilities } from '@nextcloud/capabilities'
+import { showInfo } from '@nextcloud/dialogs'
+
+import { SESSION } from '../constants.js'
+import { setSessionState } from '../services/participantsService.js'
+import { useStore } from './useStore.js'
+
+const supportSessionState = getCapabilities()?.spreed?.features?.includes('session-state')
+const INACTIVE_TIME_MS = 3 * 60 * 1000
+/**
+ * Check whether the current session is active or not:
+ * - tab or browser window was moved to background or minimized
+ * - there was not movements within tab window for a long time
+ *
+ * @return {boolean|undefined}
+ */
+export function useActiveSession() {
+	if (!supportSessionState) {
+		return supportSessionState
+	}
+
+	const store = useStore()
+	const token = computed(() => store.getters.getToken())
+	const windowIsVisible = computed(() => store.getters.windowIsVisible())
+
+	const inactiveTimer = ref(null)
+	const currentState = ref(SESSION.STATE.ACTIVE)
+
+	watch(token, () => {
+		// Joined conversation has active state by default
+		currentState.value = SESSION.STATE.ACTIVE
+	})
+
+	watch(windowIsVisible, (value) => {
+		// Change state if tab is hidden or minimized
+		// FIXME upstream: Notifications app doesn't work in background
+		if (value) {
+			setSessionAsActive()
+		} else {
+			setSessionAsInactive()
+		}
+	})
+
+	onBeforeMount(() => {
+		window.addEventListener('focus', handleWindowFocus)
+		window.addEventListener('blur', handleWindowFocus)
+	})
+
+	onBeforeUnmount(() => {
+		window.removeEventListener('focus', handleWindowFocus)
+		window.removeEventListener('blur', handleWindowFocus)
+	})
+
+	const setSessionAsActive = async () => {
+		if (currentState.value === SESSION.STATE.ACTIVE) {
+			return
+		}
+		clearTimeout(inactiveTimer.value)
+		inactiveTimer.value = null
+		currentState.value = SESSION.STATE.ACTIVE
+
+		try {
+			await setSessionState(token.value, SESSION.STATE.ACTIVE)
+		} catch (error) {
+			console.error(error)
+		}
+	}
+
+	const setSessionAsInactive = async () => {
+		if (currentState.value === SESSION.STATE.INACTIVE) {
+			return
+		}
+		clearTimeout(inactiveTimer.value)
+		inactiveTimer.value = null
+		currentState.value = SESSION.STATE.INACTIVE
+
+		try {
+			await setSessionState(token.value, SESSION.STATE.INACTIVE)
+			showInfo(t('spreed', 'Session has been marked as inactive'))
```

Comment:
console.info

Not sure about `showInfo` might be helpful if you see it on another screen though.

Reply:
Looks like this. It will provide at least some info, why you start to receive notifications from the chat, which is currently opened. But for sure, we could remove this and leave only `console.info`

![image](https://github.com/nextcloud/spreed/assets/93392545/7d19ad3e-cf59-451e-bfea-972242b57655)

Reply:
Let's do both